dali/kernels/slice/slice_cpu.h

```diff
@@ -28,31 +29,238 @@ namespace kernels {
+  int64_t pad_before = std::max(0l, std::min(out_extent, -anchor));
+  int64_t to_copy =
+      std::max(0l, std::min(in_extent - std::max(0l, anchor), out_extent - pad_before));
+  int64_t pad_after = out_extent - pad_before - to_copy;
+  return std::tuple<int64_t, int64_t, int64_t>{pad_before, to_copy, pad_after};
+}
+
+/**
+ * @brief Optimized special case for the last two dimensions whith channel-last configuration
+ */
+template <typename OutputType, typename InputType, int Dims, bool OutOfBounds, bool NeedPad,
+          int DimsLeft>
+void SliceKernelImplChannelLast(OutputType *output,
+                                const InputType *input,
+                                const TensorShape<Dims> &in_strides,
+                                const TensorShape<Dims> &out_strides,
+                                const TensorShape<Dims> &anchor,
+                                const TensorShape<Dims> &in_shape,
+                                const TensorShape<Dims> &out_shape,
+                                const OutputType *fill_values,
+                                int channel_dim,
+                                std::integral_constant<int, DimsLeft>,
+                                std::integral_constant<bool, OutOfBounds>,
+                                std::integral_constant<bool, NeedPad>) {
+  static_assert(DimsLeft == 2);
+  constexpr auto d = Dims - 2;  // NOLINT
+  assert(channel_dim == Dims - 1);
+  int64_t out_nchannels = out_shape[channel_dim];
+  int64_t in_nchannels = in_shape[channel_dim];
+  int64_t npixels = out_shape[Dims - 2];
+
+  if (NeedPad) {
+    // If the whole row is out of bounds, just fill
+    if (OutOfBounds) {
+      Fill(output, fill_values, npixels, out_nchannels);
+      return;
+    }
+
+    // Calculate number of pixels to pad on the left and right, and the number of pixels to be
+    // copied
+    int64_t pad_pixels_before, copy_pixels, pad_pixels_after;
+    std::tie(pad_pixels_before, copy_pixels, pad_pixels_after) =
+        CalcPadCopyExtents(anchor[d], in_shape[d], out_shape[d]);
+
+    // Padding pixels on the left, if needed
+    if (pad_pixels_before > 0) {
+      Fill(output, fill_values, pad_pixels_before, out_nchannels);
+      output += pad_pixels_before * out_strides[d];
+    }
+
+    // If the anchor is positive, increase the input pointer
+    if (anchor[d] > 0)
+      input += anchor[d] * in_strides[d];
+
+    // Calculate number of channels to pad on the left, right and the number of channels to be
+    // copied
+    int64_t pad_channels_before, copy_channels, pad_channels_after;
+    std::tie(pad_channels_before, copy_channels, pad_channels_after) =
+        CalcPadCopyExtents(anchor[channel_dim], in_nchannels, out_nchannels);
+    int64_t anchor_channel_in = std::max(0l, anchor[channel_dim]);
+    // Copy pixels with potential padding on the channel dimension
+    for (int64_t i = 0; i < copy_pixels; i++) {
```

Comment:
I think that for a (likely common) case when not padding channels, we're missing a fast track.